hoomd/md/pytest/test_pair.py

```diff
@@ -0,0 +1,751 @@
+
+    fourier_arg_dict = {'a': [[0.5, 1.0, 1.5],
+                              [.05, .1, .15],
+                              [.005, .01, .015]],
+                        'b': [[0.25, 0.034, 0.76],
+                              [0.36, 0.12, 0.65],
+                              [0.78, 0.04, 0.98]]}
+    fourier_valid_param_dicts = _make_valid_param_dicts(fourier_arg_dict)
+    valid_params_list.append(paramtuple(hoomd.md.pair.Fourier,
+                                        dict(zip(combos,
+                                                 fourier_valid_param_dicts)),
+                                        {}))
+
+    slj_arg_dict = {'sigma': [0.5, 1.0, 1.5],
+                    'epsilon': [0.0005, 0.001, 0.0015]}
+    slj_valid_param_dicts = _make_valid_param_dicts(slj_arg_dict)
+    valid_params_list.append(paramtuple(hoomd.md.pair.SLJ,
+                                        dict(zip(combos,
+                                                 slj_valid_param_dicts)),
+                                        {}))
+
+    dpd_arg_dict = {'A': [0.5, 1.0, 1.5],
+                    'gamma': [0.0005, 0.001, 0.0015]}
+    dpd_valid_param_dicts = _make_valid_param_dicts(dpd_arg_dict)
+    valid_params_list.append(paramtuple(hoomd.md.pair.DPD,
+                                        dict(zip(combos,
+                                                 dpd_valid_param_dicts)),
+                                        {"kT": 2}))
+
+    dpdlj_arg_dict = {'sigma': [0.5, 1.0, 1.5],
+                      'epsilon': [0.0005, 0.001, 0.0015],
+                      'gamma': [0.034, 33.2, 1.2]}
+    dpdlj_valid_param_dicts = _make_valid_param_dicts(dpdlj_arg_dict)
+
+    valid_params_list.append(paramtuple(hoomd.md.pair.DPDLJ,
+                                        dict(zip(combos,
+                                                 dpdlj_valid_param_dicts)),
+                                        {"kT": 1}))
+
+    dlvo_arg_dict = {'kappa': [1.0, 2.0, 5.0],
+                     'Z': [0.1, 0.5, 2.0],
+                     'A': [0.1, 0.5, 2.0]}
+    dlvo_valid_param_dicts = _make_valid_param_dicts(dlvo_arg_dict)
+    valid_params_list.append(paramtuple(hoomd.md.pair.DLVO,
+                                        dict(zip(combos,
+                                                 dlvo_valid_param_dicts)),
+                                        {}))
+    return valid_params_list
+
+
+@pytest.fixture(scope="function", params=_valid_params(), ids=(lambda x: x[0].__name__))
+def valid_params(request):
+    return deepcopy(request.param)
+
+def test_valid_params(valid_params):
+    pot = valid_params.pair_potential(**valid_params.extra_args,
+                                      nlist=hoomd.md.nlist.Cell(),
+                                      r_cut=2.5,
+                                      mode='none')
+    for pair in valid_params.pair_potential_params:
+        pot.params[pair] = valid_params.pair_potential_params[pair]
```

Comment:
You should be able to just set it to the entire dictionary assuming the dictionary is `{('a', 'a'): params},...}`. No need for the for loop.